### ISSUE TITLE
Test files are hanging when run from our cypress runner

### ISSUE THREAD

**Issue by georgesequeira** [cypress-io/cypress#2392]
### Current behavior:
Image: cypress/base:8
Version: 3.1.0
Browsers: Chrome + Electron
Environment: Locally in Docker and in CircleCI

When run in parallel across multiple containers, we are experiencing a number of tests hanging with the endless lines of 
`cypress:server:timers child sending timer id `

Full stack trace
```
  Running: opportunity_finder/load_opportunity_finder.js...                                (3 of 4)

cypress:server:run about to run spec { spec: { name: 'opportunity_finder/load_opportunity_finder.js', relative: 'cypress/tmp/opportunity_finder/load_opportunity_finder.js', absolute: '/e2etests/cypress/tmp/opportunity_finder/load_opportunity_finder.js' }, isHeadless: true, browserName: 'electron' } +12ms

cypress:server:run waiting for socket connection... { id: 'mcabq' } +1ms

cypress:server:openproject resetting project state, preparing to launch browser +1ms

cypress:server:project resetting project instance /e2etests +0ms

cypress:server:server Setting remoteAuth undefined +1ms

cypress:server:server Setting remoteOrigin http://localhost:33647 +0ms

cypress:server:server Setting remoteStrategy file +1ms

cypress:server:server Setting remoteHostAndPort null +0ms

cypress:server:server Setting remoteDocDomain localhost +0ms

cypress:server:server Setting remoteFileServer http://localhost:33447 +0ms

cypress:server:server Getting remote state: { auth: undefined, props: null, origin: 'http://localhost:33647', strategy: 'file', visiting: false, domainName: 'localhost', fileServer: 'http://localhost:33447' } +0ms

cypress:server:timers queuing timer id 624 after 30000 ms +1ms

cypress:server:timers child received timer id 624 +202ms

cypress:server:openproject launching browser: electron, spec: cypress/tmp/opportunity_finder/load_opportunity_finder.js +6ms

cypress:server:browsers opening browser electron +0ms

cypress:server:browsers:electron open { browserName: 'electron', url: 'http://localhost:33647/__/#/tests/integration/opportunity_finder/load_opportunity_finder.js' } +0ms

cypress:server:saved_state noop saved state +0ms

cypress:server:browsers:electron debugger detached due to { reason: 'target closed' } +23ms

cypress:server:browsers:electron received saved state {} +13ms

cypress:server:browsers:electron browser window options { width: 1280, height: 720, show: false, automationMiddleware: { onAfterResponse: [Function] }, projectRoot: '/e2etests', browsers: [ { name: 'electron', displayName: 'Electron', version: '59.0.3071.115', path: '', majorVersion: '59', info: 'Electron is the default browser that comes with Cypress. This is the browser that runs in headless mode. Selecting this browser is useful when debugging. The version number indicates the underlying Chromium version that Electron uses.' } ], proxyUrl: 'http://localhost:33647', userAgent: null, proxyServer: 'http://localhost:33647', socketIoRoute: '/__socket.io', chromeWebSecurity: true, url: 'http://localhost:33647/__/#/tests/integration/opportunity_finder/load_opportunity_finder.js', isTextTerminal: true, browser: { name: 'electron', displayName: 'Electron', version: '59.0.3071.115', path: '', majorVersion: '59', info: 'Electron is the default browser that comes with Cypress. This is the browser that runs in headless mode. Selecting this browser is useful when debugging. The version number indicates the underlying Chromium version that Electron uses.', isHeadless: true, isHeaded: false }, x: null, y: null, devTools: false, minWidth: 100, minHeight: 100, contextMenu: true, partition: 'persist:run-1509', trackState: { width: 'browserWidth', height: 'browserHeight', x: 'browserX', y: 'browserY', devTools: 'isBrowserDevToolsOpen' }, frame: true, recordFrameRate: null, onPaint: null, webPreferences: { partition: null, chromeWebSecurity: true, nodeIntegration: false, backgroundThrottling: false } } +8ms

cypress:server:plugins plugin event registered? { event: 'before:browser:launch', isRegistered: false } +8ms

cypress:server:server Getting remote state: { auth: undefined, props: null, origin: 'http://localhost:33647', strategy: 'file', visiting: false, domainName: 'localhost', fileServer: 'http://localhost:33447' } +8ms

cypress:server:proxy handling proxied request { url: '/api/clicks?args=', proxiedUrl: 'https://dev.bluecore.com/api/clicks?args=', headers: { host: 'dev.bluecore.com', connection: 'keep-alive', 'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/3.1.0 Chrome/59.0.3071.115 Electron/1.8.2 Safari/537.36', accept: 'image/webp,image/apng,image/*,*/*;q=0.8', referer: 'https://dev.bluecore.com/email_proofs', 'accept-encoding': 'gzip, deflate', 'accept-language': 'en-US', cookie: '*********' }, remoteState: { auth: undefined, props: null, origin: 'http://localhost:33647', strategy: 'file', visiting: false, domainName: 'localhost', fileServer: 'http://localhost:33447' } } +0ms

cypress:server:timers queuing timer id 625 after 60000 ms +7ms

cypress:server:server Got CONNECT request from dev.bluecore.com:443 +2ms

cypress:https-proxy Writing socket connection headers for URL: dev.bluecore.com:443 +520ms

cypress:server:server Got CONNECT request from dev.bluecore.com:443 +2ms

cypress:https-proxy Writing socket connection headers for URL: dev.bluecore.com:443 +2ms

cypress:server:browsers:electron launching browser window to url: http://localhost:33647/__/#/tests/integration/opportunity_finder/load_opportunity_finder.js +1ms

cypress:server:timers child received timer id 625 +81ms

cypress:server:browsers:electron debugger attached +19ms

cypress:server:browsers:electron clearing cache +1ms

cypress:server:server HTTPS request does not match URL: https://dev.bluecore.com:443 with props: null +8ms

cypress:https-proxy Making direct connection to dev.bluecore.com:443 +30ms

cypress:server:server HTTPS request does not match URL: https://dev.bluecore.com:443 with props: null +3ms

cypress:https-proxy Making direct connection to dev.bluecore.com:443 +3ms

cypress:server:timers queuing timer id 626 after 30000 ms +4ms

cypress:server:timers child received timer id 626 +32ms

cypress:server:browsers browser opened +25ms

cypress:server:timers queuing timer id 627 after 1000 ms +1ms

cypress:server:timers child received timer id 627 +30ms

cypress:server:timers clearing timer id 626 from queue { '619': { args: [], ms: 85000, cb: [Function] }, '624': { args: [], ms: 30000, cb: [Function: timeoutTimeout] }, '625': { args: [], ms: 60000, cb: [Function] }, '626': { args: [], ms: 30000, cb: [Function: bound cleanupWebsocketResources] }, '627': { args: [], ms: 1000, cb: [Function] } } +31ms

cypress:server:timers clearing timer id 619 from queue { '619': { args: [], ms: 85000, cb: [Function] }, '624': { args: [], ms: 30000, cb: [Function: timeoutTimeout] }, '625': { args: [], ms: 60000, cb: [Function] }, '627': { args: [], ms: 1000, cb: [Function] } } +1ms

cypress:server:timers clearing timer id 619 from queue { '624': { args: [], ms: 30000, cb: [Function: timeoutTimeout] }, '625': { args: [], ms: 60000, cb: [Function] }, '627': { args: [], ms: 1000, cb: [Function] } } +2ms

cypress:server:timers clearing timer id 625 from queue { '624': { args: [], ms: 30000, cb: [Function: timeoutTimeout] }, '625': { args: [], ms: 60000, cb: [Function] }, '627': { args: [], ms: 1000, cb: [Function] } } +2ms

cypress:server:routes Serving Cypress front-end by requested URL: /__/ +54ms

cypress:server:server Getting remote state: { auth: undefined, props: null, origin: 'http://localhost:33647', strategy: 'file', visiting: false, domainName: 'localhost', fileServer: 'http://localhost:33447' } +0ms

cypress:server:runner serving runner index.html with config { version: '3.1.0', platform: 'linux', arch: 'x64', projectName: 'e2etests' } +1ms

cypress:server:proxy redirecting to new url { status: 302, url: 'https://dev.bluecore.com/images/transparent.gif?utm_campaign=edc&utm_source=tm' } +52ms

cypress:server:browsers:electron console message: { column: 17, level: 'debug', line: 131836, source: 'console-api', text: 'Download the React DevTools for a better development experience: https://fb.me/react-devtools', url: 'http://localhost:33647/__cypress/runner/cypress_runner.js' } +787ms

cypress:server:timers child sending timer id 627 +1s

cypress:server:timers clearing timer id 627 from queue { '624': { args: [], ms: 30000, cb: [Function: timeoutTimeout] }, '627': { args: [], ms: 1000, cb: [Function] } } +77ms

cypress:server:server Got CONNECT request from localhost:33647 +210ms

cypress:https-proxy Writing socket connection headers for URL: localhost:33647 +1s

cypress:server:server HTTPS request does not match URL: https://localhost:33647 with props: null +4ms

cypress:https-proxy Making direct connection to localhost:33647 +4ms

cypress:server:server Got UPGRADE request from /__socket.io/?EIO=3&transport=websocket +4ms

cypress:server:timers queuing timer id 628 after 85000 ms +3ms

cypress:server:timers child received timer id 628 +223ms

cypress:server:socket socket connected +2ms

cypress:server:timers clearing timer id 628 from queue { '624': { args: [], ms: 30000, cb: [Function: timeoutTimeout] }, '628': { args: [], ms: 85000, cb: [Function] } } +93ms

cypress:server:timers queuing timer id 629 after 85000 ms +1ms

cypress:server:timers child received timer id 629 +96ms

cypress:server:timers clearing timer id 629 from queue { '624': { args: [], ms: 30000, cb: [Function: timeoutTimeout] }, '629': { args: [], ms: 85000, cb: [Function] } } +45ms

cypress:server:timers queuing timer id 630 after 85000 ms +0ms

cypress:server:timers queuing timer id 631 after 1000 ms +1ms

cypress:server:timers child received timer id 630 +45ms

cypress:server:timers clearing timer id 631 from queue { '624': { args: [], ms: 30000, cb: [Function: timeoutTimeout] }, '630': { args: [], ms: 85000, cb: [Function] }, '631': { args: [], ms: 1000, cb: [Function: timeoutTimeout] } } +0ms

cypress:server:timers child received timer id 631 +1ms

cypress:server:browsers:electron console message: { column: 32, level: 'log', line: 140313, source: 'console-api', text: 'console.clear', url: 'http://localhost:33647/__cypress/runner/cypress_runner.js' } +95ms

cypress:server:timers clearing timer id 630 from queue { '624': { args: [], ms: 30000, cb: [Function: timeoutTimeout] }, '630': { args: [], ms: 85000, cb: [Function] } } +0ms

cypress:server:timers queuing timer id 632 after 85000 ms +1ms

cypress:server:timers child received timer id 632 +95ms

cypress:server:run got socket connection { id: 'mcabq' } +1ms

cypress:server:timers clearing timer id 624 from queue { '624': { args: [], ms: 30000, cb: [Function: timeoutTimeout] }, '632': { args: [], ms: 85000, cb: [Function] } } +1ms

cypress:server:server Getting remote state: { auth: undefined, props: null, origin: 'http://localhost:33647', strategy: 'file', visiting: false, domainName: 'localhost', fileServer: 'http://localhost:33447' } +86ms

cypress:server:timers clearing timer id 632 from queue { '632': { args: [], ms: 85000, cb: [Function] } } +1ms

cypress:server:timers queuing timer id 633 after 85000 ms +0ms

cypress:server:timers child received timer id 633 +97ms

cypress:server:timers clearing timer id 633 from queue { '633': { args: [], ms: 85000, cb: [Function] } } +18ms

cypress:server:timers queuing timer id 634 after 85000 ms +0ms

cypress:server:timers child received timer id 634 +10ms

cypress:server:socket watch test file 'integration/opportunity_finder/load_opportunity_finder.js' +1ms

cypress:server:socket will watch test file path 'cypress/tmp/opportunity_finder/load_opportunity_finder.js' +0ms

cypress:server:preprocessor getFile /e2etests/cypress/tmp/opportunity_finder/load_opportunity_finder.js +0ms

cypress:server:appdata path: /root/.config/Cypress/cy/production/projects/e2etests-a1cfd454f206d4e35cda96969d111cab/bundles/cypress/tmp/opportunity_finder/load_opportunity_finder.js +1ms

cypress:server:plugins plugin event registered? { event: 'file:preprocessor', isRegistered: true } +1ms

cypress:server:plugins execute plugin event 'file:preprocessor' with args: EventEmitter { domain: null, _events: { rerun: [Function] }, _eventsCount: 1, _maxListeners: undefined, filePath: '/e2etests/cypress/tmp/opportunity_finder/load_opportunity_finder.js', shouldWatch: false, outputPath: '/root/.config/Cypress/cy/production/projects/e2etests-a1cfd454f206d4e35cda96969d111cab/bundles/cypress/tmp/opportunity_finder/load_opportunity_finder.js' } undefined undefined +0ms

cypress:browserify get: /e2etests/cypress/tmp/opportunity_finder/load_opportunity_finder.js +2s

cypress:browserify input: /e2etests/cypress/tmp/opportunity_finder/load_opportunity_finder.js +0ms

cypress:browserify output: /root/.config/Cypress/cy/production/projects/e2etests-a1cfd454f206d4e35cda96969d111cab/bundles/cypress/tmp/opportunity_finder/load_opportunity_finder.js +0ms

cypress:browserify browserifyOptions { extensions: [ '.js', '.jsx', '.coffee', '.cjsx' ], transform: [ [ '/root/.cache/Cypress/3.1.0/Cypress/resources/app/packages/server/node_modules/@cypress/browserify-preprocessor/cjsxify.js', [Object] ], [ '/root/.cache/Cypress/3.1.0/Cypress/resources/app/packages/server/node_modules/babelify/index.js', [Object] ] ], plugin: [], cache: {}, packageCache: {}, entries: [ '/e2etests/cypress/tmp/opportunity_finder/load_opportunity_finder.js' ] }: +0ms

cypress:browserify making bundle /root/.config/Cypress/cy/production/projects/e2etests-a1cfd454f206d4e35cda96969d111cab/bundles/cypress/tmp/opportunity_finder/load_opportunity_finder.js +15ms

cypress:server:controllers:spec request for { spec: 'cypress/support/index.js' } +24ms

cypress:server:preprocessor getFile /e2etests/cypress/support/index.js +1ms

cypress:server:plugins plugin event registered? { event: 'file:preprocessor', isRegistered: true } +0ms

cypress:server:preprocessor headless and already processed +0ms

cypress:server:controllers:spec request for { spec: 'cypress/tmp/opportunity_finder/load_opportunity_finder.js' } +2ms

cypress:server:preprocessor getFile /e2etests/cypress/tmp/opportunity_finder/load_opportunity_finder.js +0ms

cypress:server:plugins plugin event registered? { event: 'file:preprocessor', isRegistered: true } +0ms

cypress:server:preprocessor headless and already processed +1ms

cypress:server:controllers:spec sending spec { filePath: '/root/.config/Cypress/cy/production/projects/e2etests-a1cfd454f206d4e35cda96969d111cab/bundles/cypress/support/index.js' } +1ms

cypress:server:timers child sending timer id 451 +75ms

cypress:browserify finished bundling: /root/.config/Cypress/cy/production/projects/e2etests-a1cfd454f206d4e35cda96969d111cab/bundles/cypress/tmp/opportunity_finder/load_opportunity_finder.js +110ms

cypress:server:controllers:spec sending spec { filePath: '/root/.config/Cypress/cy/production/projects/e2etests-a1cfd454f206d4e35cda96969d111cab/bundles/cypress/tmp/opportunity_finder/load_opportunity_finder.js' } +99ms

cypress:server:timers clearing timer id 634 from queue { '634': { args: [], ms: 85000, cb: [Function] } } +21ms

cypress:server:timers queuing timer id 635 after 85000 ms +2ms

cypress:server:timers child received timer id 635 +81ms

cypress:server:timers clearing timer id 635 from queue { '635': { args: [], ms: 85000, cb: [Function] } } +20ms

cypress:server:timers queuing timer id 636 after 85000 ms +1ms

cypress:server:timers child received timer id 636 +20ms

cypress:server:project received runnables { id: 'r1', title: '', root: true, type: 'suite', tests: [], suites: [ { id: 'r2', title: 'Load Opportunity Finder page and assert components are there', root: false, type: 'suite', tests: [Array], suites: [] } ] } +3ms

cypress:server:reporter trying to load reporter: mocha-multi-reporters +2ms

cypress:server:reporter trying to require local reporter with path: /e2etests/mocha-multi-reporters +2ms

cypress:server:reporter trying to require local reporter with path: /e2etests/node_modules/mocha-multi-reporters +1ms


cypress:server:timers clearing timer id 636 from queue { '636': { args: [], ms: 85000, cb: [Function] } } +41ms

cypress:server:timers queuing timer id 637 after 85000 ms +0ms

cypress:server:project onMocha start +1ms

cypress:server:timers child received timer id 637 +49ms

cypress:server:reporter got mocha event 'start' with args: [ { start: '2018-08-22T02:50:18.044Z' } ] +0ms


cypress:server:timers clearing timer id 637 from queue { '637': { args: [], ms: 85000, cb: [Function] } } +51ms

cypress:server:timers queuing timer id 638 after 85000 ms +0ms

cypress:server:project onMocha suite +1ms

cypress:server:reporter got mocha event 'suite' with args: [ { id: 'r1', title: '', root: true, type: 'suite' } ] +0ms

cypress:server:timers child received timer id 638 +52ms

cypress:server:timers clearing timer id 638 from queue { '638': { args: [], ms: 85000, cb: [Function] } } +11ms

cypress:server:timers queuing timer id 639 after 85000 ms +1ms

cypress:server:project onMocha suite +2ms

cypress:server:timers child received timer id 639 +14ms

cypress:server:reporter got mocha event 'suite' with args: [ { id: 'r2', title: 'Load Opportunity Finder page and assert components are there', root: false, type: 'suite' } ] +1ms
  Load Opportunity Finder page and assert components are there

cypress:server:timers clearing timer id 639 from queue { '639': { args: [], ms: 85000, cb: [Function] } } +3ms

cypress:server:timers queuing timer id 640 after 85000 ms +1ms

cypress:server:project onMocha hook +2ms

cypress:server:timers child received timer id 640 +7ms

cypress:server:reporter got mocha event 'hook' with args: [ { id: 'r3', title: '"before all" hook', hookName: 'before all', hookId: 'h1', body: 'function () {\n    cy.loginByOkta();\n    cy.bluecoreVisit(\'/admin/opportunities/:partner\', {\n      urlParams: {\n        partner: \'qa_bluecore_3\'\n      }\n    });\n    cy.server();\n  }', type: 'hook' } ] +1ms

cypress:server:timers clearing timer id 640 from queue { '640': { args: [], ms: 85000, cb: [Function] } } +2ms

cypress:server:timers queuing timer id 641 after 85000 ms +1ms

cypress:server:timers child received timer id 641 +5ms

cypress:server:socket automation:request 'get:cookies' { domain: 'localhost' } +1ms

cypress:server:timers clearing timer id 641 from queue { '641': { args: [], ms: 85000, cb: [Function] } } +99ms

cypress:server:timers queuing timer id 642 after 85000 ms +0ms

cypress:server:timers child received timer id 642 +99ms

cypress:server:socket backend:request { eventName: 'resolve:url', args: [ 'https://dev.bluecore.com/admin/dashboard/qa_bluecore_3', { failOnStatusCode: false, auth: null } ] } +0ms

cypress:server:server resolving visit { url: 'https://dev.bluecore.com/admin/dashboard/qa_bluecore_3', headers: { host: 'localhost:33647', connection: 'Upgrade', pragma: 'no-cache', 'cache-control': 'no-cache', upgrade: 'websocket', origin: 'http://localhost:33647', 'sec-websocket-version': '13', 'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/3.1.0 Chrome/59.0.3071.115 Electron/1.8.2 Safari/537.36', 'accept-encoding': 'gzip, deflate', 'accept-language': 'en-US', 'sec-websocket-key': 'kpTgpcn5hqgCsNYudzJDEw==', 'sec-websocket-extensions': 'permessage-deflate; client_max_window_bits' }, options: { failOnStatusCode: false, auth: null } } +0ms

cypress:server:server Getting remote state: { auth: undefined, props: null, origin: 'http://localhost:33647', strategy: 'file', visiting: false, domainName: 'localhost', fileServer: 'http://localhost:33447' } +1ms

cypress:server:timers queuing timer id 643 after 60000 ms +12ms

cypress:server:timers child received timer id 643 +14ms

cypress:server:timers clearing timer id 643 from queue { '642': { args: [], ms: 85000, cb: [Function] }, '643': { args: [], ms: 60000, cb: [Function] } } +221ms

cypress:server:timers child sending timer id 631 +386ms

cypress:server:server received response for resolving url { isOkStatusCode: true, isHtml: true, contentType: 'text/html', url: 'https://dev.bluecore.com/admin/dashboard/qa_bluecore_3', status: 200, cookies: [ ******** ], statusText: 'OK', redirects: [], originalUrl: 'https://dev.bluecore.com/admin/dashboard/qa_bluecore_3' } +243ms

cypress:server:server Setting remoteAuth null +1ms

cypress:server:cors Parsed URL { port: '443', tld: 'com', domain: 'bluecore' } +2ms

cypress:server:server Setting remoteOrigin https://dev.bluecore.com +1ms

cypress:server:server Setting remoteHostAndPort { port: '443', tld: 'com', domain: 'bluecore' } +0ms

cypress:server:server Setting remoteDocDomain bluecore.com +1ms

cypress:server:server Getting remote state: { auth: null, props: { port: '443', tld: 'com', domain: 'bluecore' }, origin: 'https://dev.bluecore.com', strategy: 'http', visiting: false, domainName: 'bluecore.com', fileServer: null } +0ms

cypress:server:timers clearing timer id 642 from queue { '642': { args: [], ms: 85000, cb: [Function] } } +21ms

cypress:server:timers queuing timer id 644 after 85000 ms +0ms

cypress:server:timers child received timer id 644 +105ms

cypress:server:socket backend:request { eventName: 'preserve:run:state', args: [ { currentId: 'r3', tests: {}, startTime: '2018-08-22T02:50:18.040Z', emissions: [Object], passed: 0, failed: 0, pending: 0, numLogs: 0, autoScrollingEnabled: true, scrollTop: 0 } ] } +2ms

cypress:server:timers child sending timer id 452 +11ms

cypress:server:timers queuing timer id 645 after 30000 ms +363ms

cypress:server:timers child received timer id 645 +353ms

cypress:server:timers clearing timer id 645 from queue { '644': { args: [], ms: 85000, cb: [Function] }, '645': { args: [], ms: 30000, cb: [Function: bound cleanupWebsocketResources] } } +4ms

cypress:server:timers clearing timer id 644 from queue { '644': { args: [], ms: 85000, cb: [Function] } } +0ms

cypress:server:timers clearing timer id 644 from queue {} +5ms

cypress:server:timers child sending timer id 453 +240ms

cypress:server:timers child sending timer id 454 +509ms

cypress:server:timers child sending timer id 455 +17ms

cypress:server:timers child sending timer id 456 +5ms

cypress:server:timers child sending timer id 457 +7ms

cypress:server:timers child sending timer id 458 +7ms

cypress:server:timers child sending timer id 459 +6ms

cypress:server:timers child sending timer id 460 +371ms

cypress:server:timers child sending timer id 461 +128ms

cypress:server:timers child sending timer id 462 +248ms

cypress:server:timers child sending timer id 463 +4ms

cypress:server:timers child sending timer id 464 +5ms

cypress:server:timers child sending timer id 465 +368ms

cypress:server:timers child sending timer id 466 +543ms

cypress:server:timers child sending timer id 467 +4ms

cypress:server:timers child sending timer id 468 +6ms

cypress:server:timers child sending timer id 469 +1s

..
...
....

cypress:server:timers child sending timer id 573 +540ms

cypress:server:timers child sending timer id 574 +379ms

cypress:server:timers child sending timer id 575 +1s

cypress:server:timers child sending timer id 576 +34ms

cypress:server:timers child sending timer id 512 +71ms

cypress:server:timers child sending timer id 578 +159ms

cypress:server:timers child sending timer id 579 +1s

cypress:server:timers child sending timer id 580 +61ms

cypress:server:timers child sending timer id 581 +601ms

cypress:server:timers child sending timer id 582 +4s

cypress:server:timers child sending timer id 583 +261ms

cypress:server:timers child sending timer id 584 +245ms

cypress:server:timers child sending timer id 585 +134ms

cypress:server:timers child sending timer id 586 +359ms

cypress:server:timers child sending timer id 587 +37ms

cypress:server:timers child sending timer id 588 +40ms

cypress:server:timers child sending timer id 589 +407ms

cypress:server:timers child sending timer id 590 +233ms

cypress:server:timers child sending timer id 591 +67ms

cypress:server:timers child sending timer id 592 +32ms

cypress:server:timers child sending timer id 593 +104ms

cypress:server:timers child sending timer id 594 +28ms

cypress:server:timers child sending timer id 595 +82ms

cypress:server:timers child sending timer id 596 +39ms

cypress:server:timers child sending timer id 597 +134ms

cypress:server:timers child sending timer id 598 +49ms

cypress:server:timers child sending timer id 599 +3ms

cypress:server:timers child sending timer id 600 +190ms

cypress:server:timers child sending timer id 601 +1s

cypress:server:timers child sending timer id 602 +319ms

cypress:server:timers child sending timer id 603 +4s

cypress:server:timers child sending timer id 605 +111ms

cypress:server:timers child sending timer id 606 +59ms

cypress:server:timers child sending timer id 607 +61ms

cypress:server:timers child sending timer id 608 +13ms

cypress:server:timers child sending timer id 609 +47ms

cypress:server:timers child sending timer id 610 +99ms

cypress:server:timers child sending timer id 611 +45ms

cypress:server:timers child sending timer id 614 +151ms

cypress:server:timers child sending timer id 615 +17ms

cypress:server:timers child sending timer id 616 +19ms

cypress:server:timers child sending timer id 620 +95ms

cypress:server:timers child sending timer id 621 +38ms

cypress:server:timers child sending timer id 622 +9ms

cypress:server:timers child sending timer id 623 +149ms

cypress:server:timers child sending timer id 625 +283ms

cypress:server:timers child sending timer id 643 +2s

cypress:server:timers child sending timer id 577 +8s

cypress:server:timers child sending timer id 604 +14s

cypress:server:timers child sending timer id 612 +433ms

cypress:server:timers child sending timer id 613 +57ms

cypress:server:timers child sending timer id 617 +89ms

cypress:server:timers child sending timer id 618 +41ms

cypress:server:timers child sending timer id 619 +18ms

cypress:server:timers child sending timer id 628 +2s

cypress:server:timers child sending timer id 629 +94ms

cypress:server:timers child sending timer id 630 +45ms

cypress:server:timers child sending timer id 632 +97ms

cypress:server:timers child sending timer id 633 +95ms

cypress:server:timers child sending timer id 634 +10ms

cypress:server:timers child sending timer id 635 +158ms

cypress:server:timers child sending timer id 636 +20ms

cypress:server:timers child sending timer id 637 +48ms

cypress:server:timers child sending timer id 638 +52ms

cypress:server:timers child sending timer id 639 +13ms

cypress:server:timers child sending timer id 640 +7ms

cypress:server:timers child sending timer id 641 +5ms

cypress:server:timers child sending timer id 642 +100ms

cypress:server:timers child sending timer id 644 +508ms
```

Our simplest test that can cause this to happen
```js
//
// bluecoreVisit
Cypress.Commands.add('bluecoreVisit', (path, options = {}) => {
  cy.visit(
    Cypress.env('VERSION_URL') +
      makeUrl(path, options.urlParams, options.urlQueryParams),
    options.cypressOptions
  );
});
//


describe('Load Opportunity Finder page and assert components are there', function() {
  before(() => {
    cy.bluecoreVisit('/admin/dashboard', {
      cypressOptions: {failOnStatusCode: false}
    });

    // clear auth cookie - we need to make sure the user is logged out
    cy.clearCookie('_bcosn');
    // go to login page
    cy.bluecoreVisit('/admin/login', {
      cypressOptions: {failOnStatusCode: false}
    });

    const account = {
      email: 'text@example.com',
      password: 'notThePassword'
    };

    cy.get('input[name=username]').type(account.email);
    cy.get('input[name=password]').type(account.password);
    cy.get('form').submit();

    cy.bluecoreVisit('/admin/opportunities');
    cy.server();
  });

  it('Loads', function() {
    // Load correct Opportunity page
    cy
      .get('h1.opportunities-page_title')
      .should('have.text', 'Opportunity');

    // Have all four Opportunities on the page
    cy
      .get(':nth-child(1) > .opportunity-summary-group_title > span')
      .should('have.text', 'New Opportunity');
    cy
      .get(':nth-child(2) > .opportunity-summary-group_title > span')
      .should('have.text', 'Creatives');
    cy
      .get(':nth-child(3) > .opportunity-summary-group_title > span')
      .should('have.text', 'Audiences');
    cy
      .get(':nth-child(4) > .opportunity-summary-group_title > span')
      .should('have.text', 'Personalization');

    // Have default Opportunities shown
    cy
      .get('.opportunities-controls_content')
      .should(
        'have.text',
        'What are my opportunites?'
      );

    // Have dropdown options
    // Opportunity types
    cy
      .get(
        '.opportunities-controls_content > :nth-child(1) > .bluecore-dropdown-button > .bluecore-dropdown-button_arrow'
      )
      .click();
    cy
      .get('.bluecore-dropdown_inner_list > div > :nth-child(1) > a')
      .should('have.text', 'overall');
    cy
      .get('.bluecore-dropdown_inner_list > div > :nth-child(2) > a')
      .should('have.text', 'new');
    cy
      .get('.bluecore-dropdown_inner_list > div > :nth-child(3) > a')
      .should('have.text', 'creative');
    cy
      .get('.bluecore-dropdown_inner_list > div > :nth-child(4) > a')
      .should('have.text', 'audience');
    cy
      .get('.bluecore-dropdown_inner_list > div > :nth-child(5) > a')
      .should('have.text', 'personalization');
    cy
      .get(
        '.opportunities-controls_content > :nth-child(1) > .bluecore-dropdown-button > .bluecore-dropdown-button_arrow'
      )
      .click();

    // Opportunities
    cy
      .get(
        '.opportunities-controls_content_group > .bluecore-dropdown-button_wrapper > .bluecore-dropdown-button > .bluecore-dropdown-button_arrow'
      )
      .click();
    cy
      .get('.bluecore-dropdown_inner_list > div > :nth-child(1) > a')
      .should('have.text', 'all');
    cy
      .get('.bluecore-dropdown_inner_list > div > :nth-child(2) > a')
      .should('have.text', 'email');
    cy
      .get('.bluecore-dropdown_inner_list > div > :nth-child(3) > a')
      .should('have.text', 'facebook');
    cy
      .get('.bluecore-dropdown_inner_list > div > :nth-child(4) > a')
      .should('have.text', 'google');
    cy
      .get('.bluecore-dropdown_inner_list > div > :nth-child(5) > a')
      .should('have.text', 'snapchat');
    cy
      .get('.bluecore-dropdown_inner_list > div > :nth-child(6) > a')
      .should('have.text', 'criteo');
    cy
      .get(
        '.opportunities-controls_content_group > .bluecore-dropdown-button_wrapper > .bluecore-dropdown-button > .bluecore-dropdown-button_arrow'
      )
      .click();

    // Can load Open, Completed, Dismissed
    cy.get('.opportunities-controls_statuses > :nth-child(1)').click();
    cy.get('.opportunities-controls_statuses > :nth-child(2)').click();
    cy.get('.opportunities-controls_statuses > :nth-child(3)').click();
  });
});
```

### Desired behavior:

No hanging if possible.

### Steps to reproduce:

Unfortunately we can't give access to our application for the reproducible set of steps but i'm curious if there is anything else i can do to provide debugging information.

### Versions
Cypress: 3.1.0
Chrome + Electron

**Comment by leifdalan** [comment]
I'm having a similar symptom (tests hanging at random times), but different cause, perhaps. I'm running in Jenkins, but this also reproduces locally.

Both in Jenkins and locally cypress is being run via `docker-compose` 3.4.

`FROM cypress/base:8`
Cypress version `3.1.0`

Here's the log from running with `DEBUG=cypress*`:

cypress_1  | 2018-08-24T20:21:30.641Z cypress:cli cli starts with arguments ["/usr/local/bin/node","/usr/local/bin/cypress","run","--spec","**/envoy/*smoke*","--reporter","junit","--reporter-options","mochaFile=results-envoy/test-report.xml"]
cypress_1  | 2018-08-24T20:21:30.644Z cypress:cli NODE_OPTIONS is not set
cypress_1  | 2018-08-24T20:21:30.644Z cypress:cli program parsing arguments
cypress_1  | 2018-08-24T20:21:30.646Z cypress:cli running Cypress
cypress_1  | 2018-08-24T20:21:30.828Z cypress:cli parsed cli options { spec: '**/envoy/*smoke*',
cypress_1  |   reporter: 'junit',
cypress_1  |   reporterOptions: 'mochaFile=results-envoy/test-report.xml' }
cypress_1  | 2018-08-24T20:21:30.831Z cypress:cli verifying Cypress app
cypress_1  | 2018-08-24T20:21:30.832Z cypress:cli checking environment variables
cypress_1  | 2018-08-24T20:21:30.846Z cypress:cli checking if executable exists /root/.cache/Cypress/3.1.0/Cypress/Cypress
cypress_1  | 2018-08-24T20:21:30.850Z cypress:cli Binary is executable? : true
cypress_1  | 2018-08-24T20:21:30.851Z cypress:cli binaryDir is  /root/.cache/Cypress/3.1.0/Cypress
cypress_1  | 2018-08-24T20:21:30.851Z cypress:cli Reading binary package.json from: /root/.cache/Cypress/3.1.0/Cypress/resources/app/package.json
cypress_1  | 2018-08-24T20:21:30.857Z cypress:cli Found binary version 3.1.0 installed in: /root/.cache/Cypress/3.1.0/Cypress
cypress_1  | 2018-08-24T20:21:30.858Z cypress:cli could not read binary_state.json file
cypress_1  | 2018-08-24T20:21:30.859Z cypress:cli {}
cypress_1  | 2018-08-24T20:21:30.859Z cypress:cli is Verified ? undefined
cypress_1  | 2018-08-24T20:21:30.859Z cypress:cli running binary verification check 3.1.0
cypress_1  | It looks like this is your first time using Cypress: 3.1.0
cypress_1  | 
cypress_1  | [20:21:30]  Verifying Cypress can run /root/.cache/Cypress/3.1.0/Cypress [started]
cypress_1  | 2018-08-24T20:21:30.865Z cypress:cli clearing out the verified version
cypress_1  | 2018-08-24T20:21:30.867Z cypress:cli running smoke test
cypress_1  | 2018-08-24T20:21:30.867Z cypress:cli using Cypress executable /root/.cache/Cypress/3.1.0/Cypress/Cypress
cypress_1  | 2018-08-24T20:21:30.867Z cypress:cli needs XVFB? true
cypress_1  | 2018-08-24T20:21:30.867Z cypress:cli Starting XVFB
cypress_1  | 2018-08-24T20:21:30.934Z cypress:cli smoke test command: /root/.cache/Cypress/3.1.0/Cypress/Cypress --smoke-test --ping=807
cypress_1  | 2018-08-24T20:21:33.677Z cypress:cli smoke test stdout "807"
cypress_1  | 2018-08-24T20:21:33.682Z cypress:cli Stopping XVFB
cypress_1  | 2018-08-24T20:21:33.776Z cypress:cli write verified: true
cypress_1  | 2018-08-24T20:21:33.776Z cypress:cli could not read binary_state.json file
cypress_1  | [20:21:33]  Verified Cypress!       /root/.cache/Cypress/3.1.0/Cypress [title changed]
cypress_1  | [20:21:33]  Verified Cypress!       /root/.cache/Cypress/3.1.0/Cypress [completed]
cypress_1  | 
cypress_1  | Opening Cypress...
cypress_1  | 2018-08-24T20:21:33.783Z cypress:cli processing run options
cypress_1  | 2018-08-24T20:21:33.783Z cypress:cli --key is not set, looking up environment variable CYPRESS_RECORD_KEY
cypress_1  | 2018-08-24T20:21:33.783Z cypress:cli run to spawn.start args ["--run-project","/automation","--spec","**/envoy/*smoke*","--reporter","junit","--reporter-options","mochaFile=results-envoy/test-report.xml"]
cypress_1  | 2018-08-24T20:21:33.784Z cypress:cli needs XVFB? true
cypress_1  | 2018-08-24T20:21:33.784Z cypress:cli Starting XVFB
cypress_1  | 2018-08-24T20:21:33.800Z cypress:cli spawning Cypress with executable: /root/.cache/Cypress/3.1.0/Cypress/Cypress
cypress_1  | 2018-08-24T20:21:33.801Z cypress:cli spawn forcing env overrides { FORCE_COLOR: '0', DEBUG_COLORS: '0', FORCE_STDIN_TTY: '0', FORCE_STDOUT_TTY: '0', FORCE_STDERR_TTY: '0' }
cypress_1  | 2018-08-24T20:21:33.802Z cypress:cli spawn args [ '--run-project', '/automation', '--spec', '**/envoy/*smoke*', '--reporter', 'junit', '--reporter-options', 'mochaFile=results-envoy/test-report.xml', '--cwd', '/automation' ] { dev: undefined, detached: false, stdio: [ 'inherit', 'inherit', 'pipe' ] }
cypress_1  | Fri, 24 Aug 2018 20:21:34 GMT cypress:ts Running without ts-node hook in environment "production"
cypress_1  | Fri, 24 Aug 2018 20:21:34 GMT cypress:server:cypress starting cypress with argv [ '/root/.cache/Cypress/3.1.0/Cypress/Cypress', '--run-project', '/automation', '--spec', '**/envoy/*smoke*', '--reporter', 'junit', '--reporter-options', 'mochaFile=results-envoy/test-report.xml', '--cwd', '/automation' ]
cypress_1  | Fri, 24 Aug 2018 20:21:34 GMT cypress:server:args argv array: [ '/root/.cache/Cypress/3.1.0/Cypress/Cypress', '--run-project', '/automation', '--spec', '**/envoy/*smoke*', '--reporter', 'junit', '--reporter-options', 'mochaFile=results-envoy/test-report.xml', '--cwd', '/automation' ]
cypress_1  | Fri, 24 Aug 2018 20:21:34 GMT cypress:server:args argv parsed: { _: [ '/root/.cache/Cypress/3.1.0/Cypress/Cypress' ], runProject: '/automation', spec: '**/envoy/*smoke*', reporter: 'junit', reporterOptions: 'mochaFile=results-envoy/test-report.xml', cwd: '/automation' }
cypress_1  | Fri, 24 Aug 2018 20:21:34 GMT cypress:server:args argv options: { _: [ '/root/.cache/Cypress/3.1.0/Cypress/Cypress' ], runProject: '/automation', spec: [ '/automation/**/envoy/*smoke*' ], cwd: '/automation', config: { reporter: 'junit', reporterOptions: { mochaFile: 'results-envoy/test-report.xml' } }, projectRoot: '/automation', run: true }
cypress_1  | Fri, 24 Aug 2018 20:21:34 GMT cypress:server:appdata path: /root/.config/Cypress/cy/production
cypress_1  | Fri, 24 Aug 2018 20:21:34 GMT cypress:server:cypress starting in mode run
cypress_1  | Fri, 24 Aug 2018 20:21:36 GMT cypress:server:appdata path: /root/.config/Cypress/cy/production/browsers
cypress_1  | Fri, 24 Aug 2018 20:21:36 GMT cypress:server:appdata path: /root/.config/Cypress/cy/production
cypress_1  | Fri, 24 Aug 2018 20:21:36 GMT cypress:server:appdata path: /root/.config/Cypress/cy/production/cache
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:timers queuing timer id 1 after 500 ms
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:timers child received timer id 1
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:timers child sending timer id 1
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:timers clearing timer id 1 from queue { '1': { args: [], ms: 500, cb: [Function] } }
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:run run mode ready with options { _: [ '/root/.cache/Cypress/3.1.0/Cypress/Cypress' ], runProject: '/automation', spec: [ '/automation/**/envoy/*smoke*' ], cwd: '/automation', config: { reporter: 'junit', reporterOptions: { mochaFile: 'results-envoy/test-report.xml' } }, projectRoot: '/automation', run: true }
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:project Project created /automation
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher checking one browser chrome
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher looking up chrome on linux platform
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher looking using command "google-chrome --version"
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher Could not detect browser Command failed: /bin/sh -c google-chrome --version
cypress_1  | /bin/sh: 1: google-chrome: not found
cypress_1  | 
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher browser chrome not installed
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher checking one browser chromium
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher looking up chromium on linux platform
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher looking using command "chromium-browser --version"
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher Could not detect browser Command failed: /bin/sh -c chromium-browser --version
cypress_1  | /bin/sh: 1: chromium-browser: not found
cypress_1  | 
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher browser chromium not installed
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher checking one browser canary
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher looking up canary on linux platform
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher looking using command "google-chrome-canary --version"
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher Could not detect browser Command failed: /bin/sh -c google-chrome-canary --version
cypress_1  | /bin/sh: 1: google-chrome-canary: not found
cypress_1  | 
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:launcher browser canary not installed
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:openproject opening project /automation
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:project opening project instance /automation
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:config setting support file /automation/cypress/support
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:config for project root /automation
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:config set support folder /automation/cypress/support
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:config setting plugins file /automation/cypress/plugins
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:config for project root /automation
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:config set pluginsFile to /automation/cypress/plugins/index.js
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:config set scaffold paths
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:scaffold scaffolded files ["cypress/integration/examples/actions.spec.js","cypress/integration/examples/aliasing.spec.js","cypress/integration/examples/assertions.spec.js","cypress/integration/examples/connectors.spec.js","cypress/integration/examples/cookies.spec.js","cypress/integration/examples/cypress_api.spec.js","cypress/integration/examples/files.spec.js","cypress/integration/examples/local_storage.spec.js","cypress/integration/examples/location.spec.js","cypress/integration/examples/misc.spec.js","cypress/integration/examples/navigation.spec.js","cypress/integration/examples/network_requests.spec.js","cypress/integration/examples/querying.spec.js","cypress/integration/examples/spies_stubs_clocks.spec.js","cypress/integration/examples/traversal.spec.js","cypress/integration/examples/utilities.spec.js","cypress/integration/examples/viewport.spec.js","cypress/integration/examples/waiting.spec.js","cypress/integration/examples/window.spec.js","cypress/fixtures/example.json","cypress/support/commands.js","cypress/support/index.js","cypress/plugins/index.js"]
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:config got file tree
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:project get saved state
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:saved_state noop saved state
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:scaffold plugins folder /automation/cypress/plugins
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:scaffold verify scaffolding in /automation/cypress/plugins
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:scaffold folder /automation/cypress/plugins already exists
cypress_1  | Fri, 24 Aug 2018 20:21:40 GMT cypress:server:plugins plugins.init /automation/cypress/plugins/index.js
cypress_1  | Fri, 24 Aug 2018 20:21:41 GMT cypress:server:plugins:child pluginsFile: /automation/cypress/plugins/index.js
cypress_1  | Fri, 24 Aug 2018 20:21:41 GMT cypress:server:plugins:child require pluginsFile
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:plugins:child run plugins function
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:plugins:child register event _get:task:body with id 0
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:plugins:child register event _get:task:keys with id 1
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:plugins:child register event file:preprocessor with id 2
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:plugins register plugins process event _get:task:body with id 0
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:plugins register event '_get:task:body'
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:plugins register plugins process event _get:task:keys with id 1
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:plugins register event '_get:task:keys'
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:plugins register plugins process event file:preprocessor with id 2
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:plugins register event 'file:preprocessor'
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:project plugin config yielded: null
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:server Server listening on port 33732
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:appdata path: /root/.config/Cypress/cy/production/proxy
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:https-proxy Created SNI HTTPS Proxy on port 35042
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:server Setting remoteAuth undefined
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:cors Parsed URL { port: '8000', tld: 'envoy', domain: '' }
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:server Setting remoteOrigin http://envoy:8000
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:server Setting remoteHostAndPort { port: '8000', tld: 'envoy', domain: '' }
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:server Setting remoteDocDomain envoy
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:server Getting remote state: { auth: undefined, props: { port: '8000', tld: 'envoy', domain: '' }, origin: 'http://envoy:8000', strategy: 'http', visiting: undefined, domainName: 'envoy', fileServer: null }
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:project project config: { chromeWebSecurity: false, baseUrl: 'http://envoy:8000', defaultCommandTimeout: 10000, execTimeout: 5000, taskTimeout: 5000, requestTimeout: 5000, responseTimeout: 5000, pageLoadTimeout: 5000, viewportWidth: 1280, viewportHeight: 800, projectRoot: '/automation', projectName: 'automation', morgan: false, isTextTerminal: true, socketId: 'g1hx8', report: true, browsers: [ { name: 'electron', displayName: 'Electron', version: '59.0.3071.115', path: '', majorVersion: '59', info: 'Electron is the default browser that comes with Cypress. This is the browser that runs in headless mode. Selecting this browser is useful when debugging. The version number indicates the underlying Chromium version that Electron uses.' } ], reporter: 'junit', reporterOptions: { mochaFile: 'results-envoy/test-report.xml' }, port: 33732, hosts: null, userAgent: null, blacklistHosts: null, clientRoute: '/__/', xhrRoute: '/xhrs/', socketIoRoute: '/__socket.io', socketIoCookie: '__socket.io', reporterRoute: '/__cypress/reporter', ignoreTestFiles: '*.hot-update.js', testFiles: '**/*.*', video: true, videoCompression: 32, videoUploadOnPasses: true, modifyObstructiveCode: true, waitForAnimations: true, animationDistanceThreshold: 5, numTestsKeptInMemory: 0, watchForFileChanges: false, trashAssetsBeforeRuns: true, autoOpen: false, fileServerFolder: '/automation', videosFolder: '/automation/videos', supportFile: '/automation/cypress/support/index.js', fixturesFolder: '/automation/cypress/fixtures', integrationFolder: '/automation/cypress/integration', screenshotsFolder: '/automation/screenshots', namespace: '__cypress', pluginsFile: '/automation/cypress/plugins/index.js', javascripts: [], env: { }, cypressEnv: 'production', parentTestsFolder: '/automation/cypress', parentTestsFolderDisplay: 'automation/cypress', supportFolder: '/automation/cypress/support', integrationExampleName: 'examples', integrationExamplePath: '/automation/cypress/integration/examples', scaffoldedFiles: [ { name: 'cypress', children: [Array] } ], state: {}, proxyUrl: 'http://localhost:33732', browserUrl: 'http://envoy:8000/__/', reporterUrl: 'http://envoy:8000/__cypress/reporter', xhrUrl: '__cypress/xhrs/' }
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:reporter trying to load reporter: junit
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:reporter junit is built-in reporter
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:project scaffolding project /automation
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:scaffold support folder /automation/cypress/support, support file /automation/cypress/support/index.js
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:scaffold verify scaffolding in /automation/cypress/support
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:scaffold folder /automation/cypress/support already exists
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:project attempt watch plugins file: /automation/cypress/plugins/index.js
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:project plugins file found? true
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:project watch plugins file
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:timers queuing timer id 2 after 0 ms
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:timers child received timer id 2
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:timers child sending timer id 2
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:timers clearing timer id 2 from queue { '2': { args: [], ms: 0, cb: [Function: clear] } }
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:timers clearing timer id 2 from queue {}
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:specs looking for test specs in the folder: /automation/cypress/integration
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:profilecleaner found 0 root level profile matches: []
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:profilecleaner found 0 profile folders: []
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:profilecleaner found 0 profile folders: []
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher checking one browser chrome
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher looking up chrome on linux platform
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher looking using command "google-chrome --version"
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher Could not detect browser Command failed: /bin/sh -c google-chrome --version
cypress_1  | /bin/sh: 1: google-chrome: not found
cypress_1  | 
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher browser chrome not installed
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher checking one browser chromium
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher looking up chromium on linux platform
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher looking using command "chromium-browser --version"
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:specs found spec file /automation/cypress/integration/envoy/smoke-tests.feature
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:specs found 1 spec files: [ { name: 'envoy/smoke-tests.feature', relative: 'cypress/integration/envoy/smoke-tests.feature', absolute: '/automation/cypress/integration/envoy/smoke-tests.feature' } ]
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:server:run found '1' specs using spec pattern '/automation/**/envoy/*smoke*': [ 'envoy/smoke-tests.feature' ]
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher Could not detect browser Command failed: /bin/sh -c chromium-browser --version
cypress_1  | /bin/sh: 1: chromium-browser: not found
cypress_1  | 
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher browser chromium not installed
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher checking one browser canary
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher looking up canary on linux platform
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher looking using command "google-chrome-canary --version"
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher Could not detect browser Command failed: /bin/sh -c google-chrome-canary --version
cypress_1  | /bin/sh: 1: google-chrome-canary: not found
cypress_1  | 
cypress_1  | Fri, 24 Aug 2018 20:21:42 GMT cypress:launcher browser canary not installed

**Comment by georgesequeira** [comment]
Hi, it seems that you are trying to use the google chrome browser, that would require using the `cypress/browsers:chrome67` docker image. Have you tried that?

**Comment by leifdalan** [comment]
@georgesequeira 

I was using correct browser args / image combinations, but I just figured out what my issue was.

For anyone else with this problem, I was running cypress inside the same `docker-compose` file as the services that it was testing.

I had the following docker volumes setup so that I could export the videos/screenshots as artifacts in CI.
```
    volumes: 
      - './automation/cypress:/automation/cypress'
      - './scripts:/scripts'
      - './envoy/videos:/automation/videos'
      - './envoy/screenshots:/automation/screenshots'
      - './envoy/results:/automation/results-envoy'
```

On the second run, on the host system mounted the previous run's artifacts, which included a `.Trash-0` folder which seemed to be very recursive in file structure. 

This would cause cypress to take a _really long time_ to start, I'm guessing because of some file system operation that it was being overwhelmed by with the leftover volume mount.

Adding a cleanup step `docker-compose down -v` in my pipeline scripts solved this problem.
First run would be fine, but when

**Comment by jennifer-shehane** [comment]
Unfortunately we have to close this issue due to inactivity. Please open a new issue if there is new information to provide concerning the original issue.
